### ISSUE TITLE
Name attributes containing a dot do not work

### ISSUE THREAD

**Issue by rok7** [aspnet/jquery-validation-unobtrusive#39]
When using nested properties, unobstrusive validation does not work.
Example: 
```
<span asp-validation-for="Customer.Phone" class="text-danger"></span>
```

The reason for this is the escapeAttributeValue call in the function onError(), which removes the dot from the field name and therefore doesn't find the container with this name
```
function onError(error, inputElement) {
   var container = $(this).find("[data-valmsg-for='" + escapeAttributeValue(inputElement[0].name) + "']"),
....

function escapeAttributeValue(value) {
        // As mentioned on http://api.jquery.com/category/selectors/
        return value.replace(/([!"#$%&'()*+,./:;<=>?@\[\\\]^`{|}~])/g, "\\\\$1");
....
```
 
### Functional impact
Validation does not work on nested properties

**Comment by kbherbert** [comment]
I just wanted to point out that I am using jQuery 2.2.4 and I am not seeing this issue. Wondering if you are seeing this as a result of some odd combination (jQuery + jQuery validation + jquery Unobtrusive)?

**Comment by rok7** [comment]
Thank you for your Response, you might be right: my combination was 
jquery: 2.2.0, jquery-validation: 1.15.0, jquery-validation-unobtrusive: 3.2.6. 
After changing jquery-validation to 1.14.0 I still have the same Problem.

Which combination are you using?

**Comment by Eilon** [comment]
@ryanbrandenburg can you try to repro and see what's up with this?

**Comment by szykov** [comment]
Hi' i'm having the same issue. If i use nested properties jquery unobtrussive stops working.
i'm using:
    "jquery": "2.2.4",
    "jquery-validation": "1.15.0",
    "jquery-validation-unobtrusive": "3.2.6",

i've created theme here also:

http://stackoverflow.com/questions/38131201/jquery-unobtrusive-stops-working-if-asp-for-has-dots-in-name

**Comment by ryanbrandenburg** [comment]
@rok7 are you using `jquery.validation.unobtrusive.js` or `jquery.validation.unobtrusive.min.js`? It appears that our *.min file for this repo is behind, and there's a fix missing from `min` that could definitely explain your problems. If you're using min please try it with the un-minified version. If that works we'll update our *.min.js, if not please upload a repro app (complete with whatever versions of these three frameworks you have) so I can dig further.

**Comment by ryanbrandenburg** [comment]
I dug at this a little more and found out that our NPM repository is also a little bit out of date. If you try a copy of `jquery.validation.unobtrusive.js` directly from github does it work? Either way we're working to get it updated and improve our process to prevent future missed versions.

**Comment by szykov** [comment]
@ryanbrandenburg just tried to copy js files from github directly and it didn't help.

validation-summary-errors and server side validation works. but client side isn't working at all.

**Comment by ryanbrandenburg** [comment]
In that case can you upload your app (or a parred down app that displays the same behavior) to a GitHub repo so that I can investigate it? Please include whatever .js files you have on disk rather than relying on NPM to bring them in.

**Comment by ryanbrandenburg** [comment]
A reminder, I'll need a full application which displays this behavior uploaded as a Github repo to debug this issue further.

**Comment by rok7** [comment]
While trying to provide an application for investigation I found the problem:
When a new ASP.Net Core application is created everything works ok. However, if in VisualStudio the file bower.json is manually modified (because jquery-validation-unobtrusive is not listed by the ‘Browse’ function, it's necessary for example to select a different version), then VisualStudio restores jquery-validation-unobtrusive with the following difference to the original file :
Original:

    `function escapeAttributeValue(value) {

        // As mentioned on http://api.jquery.com/category/selectors/

        return value.replace(/([!"#$%&'()*+,./:;<=>?@\[\\\]^`{|}~])/g, "\\$1");
    }`

VisualStudio-restored
    `function escapeAttributeValue(value) {

        // As mentioned on http://api.jquery.com/category/selectors/

        return value.replace(/([!"#$%&'()*+,./:;<=>?@\[\\\]^`{|}~])/g, "\\\\$1");
    }`
These magically inserted ‘\\’ caused my problems!!!

**Comment by ryanbrandenburg** [comment]
Those `\` were the result of an attempt to fix a different issue, but upon closer inspection that issue was caused by something else and we can revert that change. That's happening in #44.

**Comment by szykov** [comment]
@ryanbrandenburg Hi, sorry for delay. Here's the link to github repository:
https://github.com/szykov/WebServicesApp

It is my first try to upload anything to github. You can access the form by using something like this:
http://localhost:53000/spare-parts

Thanks.

**Comment by szykov** [comment]
@rok7 thanks. that's indeed solved my problem. thanks.

**Comment by ryanbrandenburg** [comment]
This should all be resolved now that #44 is merged in.
